src/parser/transformer.cpp

```diff
@@ -498,34 +498,48 @@ unique_ptr<ParsedExpression> Transformer::transformListOperatorExpression(
     CypherParser::OC_ListOperatorExpressionContext& ctx,
     unique_ptr<ParsedExpression> propertyExpression) {
     auto rawExpression = propertyExpression->getRawName() + " " + ctx.getText();
-
-    if (ctx.children[1]->getText() == ":" || ctx.children[2]->getText() == ":") {
+    if (ctx.kU_ListSliceOperatorExpression()) {
         auto listSlice =
             make_unique<ParsedFunctionExpression>(LIST_SLICE_FUNC_NAME, move(rawExpression));
         listSlice->addChild(move(propertyExpression));
-        if (ctx.children[1]->getText() == ":") {
+        if (ctx.kU_ListSliceOperatorExpression()->children[1]->getText() == ":") {
```

Comment:
I understand this code path but let's leave a comment here about parsing `[ : ]` `[ left : ]` `[ : right ]`. In future we might change grammar to make this more readable.

Reply:
Done